crates/nu-plugin-test-support/src/plugin_test.rs

```diff
@@ -345,4 +346,21 @@ impl PluginTest {
             _ => Ok(a == b),
         }
     }
+
+    /// This implements custom value comparison with `plugin.custom_value_to_base_value()` to behave
+    /// as similarly as possible to comparison in the engine.
+    ///
+    /// For non custom values, a clone of a value is returned.
+    pub fn base_value(&self, value: &Value) -> Result<Value, ShellError> {
```

Comment:
I think my main suggestion would be that this name should probably reflect either `to_base_value` or `custom_value_to_base_value` to help someone find it, and in case you end up adding the others here.

Maybe this should just take `&dyn CustomValue` to reflect the signature exactly so that they're all pretty much the same? Just thinking of least surprise here

Reply:
done!